doc/library/tensor/nnet/conv.txt

```diff
@@ -22,78 +22,62 @@
 .. moduleauthor:: LISA
 
 
-.. note::
+The recomanded user interface are:
+
+- :func:`theano.tensor.nnet.conv2d` for 2d convolution
+- :func:`theano.tensor.nnet.conv3d` for 3d convolution
 
-    As of December 2015, a new conv2d interface has been introduced.
-    :func:`nnet.conv2d <theano.tensor.nnet.conv2d>` defines an
-    abstract theano graph convolution operation
-    (:func:`nnet.abstract_conv.AbstractConv2d <theano.tensor.nnet.abstract_conv.AbstractConv2d>`)
-    that will be replaced by an actual convolution implementation during
-    the optimization phase.
+With those new interface, Theano will automatically use the fastest
+implementation in many cases. On the CPU, the implementation is a GEMM
+based one. On the GPU, there is a GEMM based and :ref:`cuDNN
+<libdoc_gpuarray_dnn>` version.
 
-    As of October 2016 (version 0.9.0dev3), there is also a conv3d interface that provides
-    a similar operation for 3D convolution. :func:`nnet.conv3d <theano.tensor.nnet.conv3d>`
-    defines the abstract theano graph convolution operation
-    :func:`nnet.abstract_conv.AbstractConv3d <theano.tensor.nnet.abstract_conv.AbstractConv3d>`.
+By default on the GPU, if cuDNN is available, it will be used,
+otherwise we will fall back to using gemm based version (slower than
+cuDNN in most cases and uses more memory). To get an error if cuDNN
+can not be used, you can supply the Theano flag ``dnn.enable=True``.
 
-    Since the abstract Op does not have any implementation, it will prevent
-    computations in the un-optimized graph, and cause problems with DebugMode,
-    test values, and when compiling with optimizer=None.
+Either cuDNN and the gemm version can be disabled using the Theano flags
+``optimizer_excluding=conv_dnn`` and ``optimizer_excluding=conv_gemm``,
+respectively. If both are disabled, it will raise an error.
 
-    By default, if :ref:`cuDNN <libdoc_gpuarray_dnn>`
-    is available, we will use it, otherwise we will fall back to using the
-    gemm version (slower than cuDNN in most cases and uses more memory).
 
-    Either cuDNN and the gemm version can be disabled using the Theano flags
-    ``optimizer_excluding=conv_dnn`` and ``optimizer_excluding=conv_gemm``,
-    respectively. In this case, we will fall back to using the legacy
-    convolution code, which is slower, but does not require extra memory.
-    To verify that cuDNN is used, you can supply the Theano flag
-    ``optimizer_including=cudnn``. This will raise an error if cuDNN is
-    unavailable.
+For the cuDNN version, there is different algorythm with different
+memory/speed trade-off. Manual selection of the right one is very
+difficult as it depend of the shapes and hardware. So it can change
+for each layer. An auto-tuning mode exist and can be activated by
+those flag: ``dnn.conv.algo_fwd=time_once``,
+``dnn.conv.algo_bwd_data=time_once`` and
+``dnn.conv.algo_bwd_filter=time_once``.
```

Comment:
For theses flages, should we recommand `time_once`, or `time_on_shape_change` instead (even if it is currently not perfect) ?

In fact, `time_once` is good if shapes don't change. But it may cause errors if shapes change and the new shape is not supported by the algorithm selected in first call.